arch/src/x86_64/mod.rs

```diff
@@ -64,6 +64,9 @@ pub const _NSIG: i32 = 65;
 pub struct EntryPoint {
     /// Address in guest memory where the guest must start execution
     pub entry_addr: GuestAddress,
+    /// Page frame number of the AMD vmsa page
+    #[cfg(feature = "sev_snp")]
+    pub vmsa_pfn: u64,
```

Comment:
Can we just override the entry_addr with vmsa_pfn in case of SEV-SNP?

Reply:
Thanks, Updated it.